src/Swarm/Game/Robot.hs

```diff
@@ -150,8 +155,14 @@ data RobotR f = RobotR
   , _tickSteps :: Int
   , _robotCreatedAt :: TimeSpec
   }
+  deriving (Generic)
 
 deriving instance Show (f RID) => Show (RobotR f)
+deriving instance FromJSON (f RID) => FromJSON (RobotR f)
+deriving instance ToJSON (f RID) => ToJSON (RobotR f)
```

Comment:
This is a bit surprising since we have the `FromJSONE` instance above.

That one requires the entity map and also does not allow specifying robot ID nor some other fields like the log.

I am not sure what this `ToJSON` instance produces, but I would like it to match the `FromJSONE`.
Though if the generated instance is reasonable maybe it could be useful and we could align them later.

Reply:
To be honest, I have no idea, I just typed on the keyboard until GHC could link the binary :-)

Reply:
Note, when I made the `FromJSONE` instances I was just interested in putting in enough functionality to be able to describe robots in the ways that I wanted in scenario description files.  But if there are fields that are still missing we can definitely add them.